### ISSUE TITLE
Add a 'Remove Magnet' button to DuckieTV's TorrentClients

### ISSUE THREAD

**Issue by garfield69** [SchizoDuckie/DuckieTV#441]
It may be useful to add a button that allows a user to remove a magnet from the TorrentHost.
This would remove only the magnet details on the TorrentHost, not any of the files downloaded, and also null out the episode's magnet entity in DuckieTV's DB Episode's table.

This would also solve js41637's question in #419
>Is there something that can be done about the torrent controls after an episode has been downloaded or its no longer being seeded. With the addition of a subtitle button and long show descriptions. The torrent controls can take up a lot of unneeded space and remain on any episode that has been downloaded.

as once the magnet is removed from the DuckieTV DB, the TorrentClient controls for that episode would not be displayed.

**Comment by Js41637** [comment]
Yeah me and @SchizoDuckie have had quite long dicussions about this that don't really lead anywhere. One of them was when someone downloads a torrent and then remove the torrent when it's finished, DuckieTV doesn't know this or do anything about it and thus the controls were still shown and the progress bar*. I wanted a way for DuckieTV to realise when a torrent no longer exists in the client and remove the magnetHash from the DB thus removing the torrent controls and progress bar from the episode event on calendar.

With the current option of Marking as Downloaded I'm not if we should hide torrent controls if it has been marked as downloaded. OR at least maybe hide them by default which a button to expand them. I do agree for a way to remove a magnetHash from an episode i'm just not sure about a good way to go about doing it.

**Comment by lamaresh** [comment]
could it be automated? IE every time you open an episode tab duckieTV check if the magnet is still present in torrent client

**Comment by SchizoDuckie** [comment]
Yes, I think we should be able to do that, as the torrentClient has a list of magnets that exists.

**Comment by garfield69** [comment]
implement and test checklist.
- [x] deluge
- [x] qBitTorrent
- [x] qBitTorrentPlus
- [x] Tixati
- [x] Transmission
- [ ] uTorrentAPI
- [x] uTorrent WEBUI
- [x] Vuze

**Comment by garfield69** [comment]
Now that I have a better understanding of how the uTorrent API works, I realise that I need to amend my WEB client commits.
ILT that the advantage that the uTorrent API Client has over the WEBUI clients, is that the former is notified when a torrent is removed from the torrentHost, while the latter has to pro-actively fetch updates from the torrentHost to find out if a torrent was removed.
So for the WEBUI clients, I need to update the baseTorrentClient torrents objects so that all the DuckieTV processes can globally become aware of removed torrents rather than just the episode-details sidepanel.

TLDR: only the uTorrent API client correctly implements this ticket, the WEBUI clients need further work.

**Comment by garfield69** [comment]
WTF? why would a change to BaseTorrentClients only work for uTorrentWEBUI and not any of the other WEB clients!!! aren't they all plumbed in the same way?
Argh. Alright. Fine. I'll come back to it when I've had a long break.
